src/vm/dllimportcallback.cpp

```diff
@@ -1416,6 +1416,7 @@ VOID UMThunkMarshInfo::RunTimeInit()
     {
         // caller pop
         m_cbRetPop = 0;
+        m_cbActualArgSize += numRegistersUsed * STACK_ELEM_SIZE;
```

Comment:
Here is a dump produced by command `COMPlus_JitDisasm="DomainBoundILStubClass:IL_STUB_ReversePInvoke"` for `DoCallBack_MarshalRefInt_Stdcall` before it crashed: 
```

; Assembly listing for method DomainBoundILStubClass:IL_STUB_ReversePInvoke(int):int
; Emitting BLENDED_CODE for generic X86 CPU
; debuggable code
; ebp based frame
; fully interruptible
; Final local variable assignments
;
;  V00 arg0         [V00,T00] (  3,   3  )     int  ->  [ebp-0x04]   do-not-enreg[H]
;  V01 loc0         [V01,T02] (  3,   3  )     int  ->  [ebp-0x08]   do-not-enreg[H]
;  V02 loc1         [V02    ] (  2,   2  )   byref  ->  [ebp-0x0C]   do-not-enreg[H] must-init
;  V03 loc2         [V03,T04] (  2,   2  )    bool  ->  [ebp-0x10]   do-not-enreg[H]
;  V04 loc3         [V04,T03] (  3,   3  )     int  ->  [ebp-0x14]   do-not-enreg[H]
;  V05 loc4         [V05,T05] (  2,   2  )     int  ->  [ebp-0x18]   do-not-enreg[H]
;  V06 tmp0         [V06    ] (  3,   3  )     int  ->  [ebp-0x24]   do-not-enreg[XH] addr-exposed
;  V07 tmp1         [V07,T01] (  2,   4  )     int  ->  [ebp-0x1C]   do-not-enreg[H]
;  V08 tmp2         [V08    ] (  2,   4  )     ref  ->  [ebp-0x20]   do-not-enreg[H] must-init
;
; Lcl frame size = 40

G_M33306_IG01:
       55           push     ebp
       8BEC         mov      ebp, esp
       83EC28       sub      esp, 40
       8945DC       mov      dword ptr [ebp-24H], eax
       33D2         xor      edx, edx
       8955F4       mov      dword ptr [ebp-0CH], edx
       8955E0       mov      dword ptr [ebp-20H], edx
       894DFC       mov      dword ptr [ebp-04H], ecx

G_M33306_IG02:
       33C9         xor      ecx, ecx
       894DF8       mov      dword ptr [ebp-08H], ecx

G_M33306_IG03:
       90           nop      
       8B4DFC       mov      ecx, bword ptr [ebp-04H]
       894DF4       mov      bword ptr [ebp-0CH], ecx
       90           nop      
       90           nop      
       90           nop      
       8B4DDC       mov      ecx, dword ptr [ebp-24H]
       8B4908       mov      ecx, dword ptr [ecx+8]
       8B09         mov      ecx, gword ptr [ecx]
       8B4904       mov      ecx, gword ptr [ecx+4]
       894DE0       mov      gword ptr [ebp-20H], ecx
       8B4DDC       mov      ecx, dword ptr [ebp-24H]
       8B4908       mov      ecx, dword ptr [ecx+8]
       8B09         mov      ecx, gword ptr [ecx]
       8B490C       mov      ecx, dword ptr [ecx+12]
       894DE4       mov      dword ptr [ebp-1CH], ecx
       90           nop      
       8B4DE0       mov      ecx, gword ptr [ebp-20H]
       8B55F4       mov      edx, bword ptr [ebp-0CH]
       8B45E4       mov      eax, dword ptr [ebp-1CH]
       FFD0         call     eax
       8945F0       mov      dword ptr [ebp-10H], eax
       8B45F0       mov      eax, dword ptr [ebp-10H]
       85C0         test     eax, eax
       0F95C0       setne    al
       0FB6C0       movzx    eax, al
       8945EC       mov      dword ptr [ebp-14H], eax
       90           nop      
       8B45EC       mov      eax, dword ptr [ebp-14H]
       8945E8       mov      dword ptr [ebp-18H], eax
       90           nop      
       90           nop      
       C745F8FFFFFF7F mov      dword ptr [ebp-08H], 0x7FFFFFFF
       E808000000   call     G_M33306_IG06
       90           nop      

G_M33306_IG04:
       8B45E8       mov      eax, dword ptr [ebp-18H]

G_M33306_IG05:
       8BE5         mov      esp, ebp
       5D           pop      ebp
       C3           ret      

G_M33306_IG06:
       83EC0C       sub      esp, 12

G_M33306_IG07:
       8B45F8       mov      eax, dword ptr [ebp-08H]
       3DFEFFFF7F   cmp      eax, 0x7FFFFFFE
       7F05         jg       SHORT G_M33306_IG08
       33C0         xor      eax, eax
       8945EC       mov      dword ptr [ebp-14H], eax

G_M33306_IG08:
       83C40C       add      esp, 12
       C3           ret      

; Total bytes of code 143, prolog size 20 for method DomainBoundILStubClass:IL_STUB_ReversePInvoke(int):int
; ============================================================


```